File: mobile/lib/models/activo.dart
class Activo {
  final int? id;
  final String? codigo;
  final String nombre;
  final int categoriaId;
  final double valor;
  final String fechaAdquisicion; // yyyy-MM-dd
  final String? proveedor;
  final String? observaciones;
  final String? estado;

  Activo({
    this.id,
    this.codigo,
    required this.nombre,
    required this.categoriaId,
    required this.valor,
    required this.fechaAdquisicion,
    this.proveedor,
    this.observaciones,
    this.estado,
  });

  factory Activo.fromJson(Map<String, dynamic> json) {
    return Activo(
      id: json['id'],
      codigo: json['codigo'],
      nombre: json['nombre'],
      categoriaId: json['categoria'] != null ? json['categoria']['id'] : 0,
      valor: (json['valor'] as num).toDouble(),
      fechaAdquisicion: json['fechaAdquisicion'] ?? '',
      proveedor: json['proveedor'],
      observaciones: json['observaciones'],
      estado: json['estado'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'nombre': nombre,
      'categoriaId': categoriaId,
      'valor': valor,
      'fechaAdquisicion': fechaAdquisicion,
      'proveedor': proveedor,
      'observaciones': observaciones,
    };
  }
}

File: backend/src/main/java/com/uagrm/activos/model/Activo.java
package com.uagrm.activos.model;

import jakarta.persistence.*;
import lombok.*;
import java.math.BigDecimal;
import java.time.LocalDate;

@Entity
@Table(name = "activo")
@Getter @Setter @NoArgsConstructor @AllArgsConstructor @Builder
public class Activo {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 30)
    private String codigo; // HU-02: codigo unico generado por el sistema

    @Column(nullable = false, length = 150)
    private String nombre;

    @ManyToOne(optional = false)
    @JoinColumn(name = "categoria_id")
    private Categoria categoria;

    @Column(nullable = false)
    private BigDecimal valor; // HU-02: no puede ser negativo ni cero

    @Column(name = "fecha_adquisicion", nullable = false)
    private LocalDate fechaAdquisicion;

    @Column(length = 150)
    private String proveedor;

    private String observaciones;

    @Enumerated(EnumType.STRING)
    @Builder.Default
    private EstadoActivo estado = EstadoActivo.SIN_ASIGNAR;

    @ManyToOne
    @JoinColumn(name = "responsable_id")
    private Usuario responsable; // HU-03

    @ManyToOne
    @JoinColumn(name = "ubicacion_id")
    private Ubicacion ubicacion; // HU-03
}
